circuit/src/ecc/native.rs

```diff
@@ -125,38 +166,34 @@ where
 		Self { x: r_x.result, y: r_y.result }
 	}
 
-	/// Scalar multiplication for given point
+	/// Scalar multiplication for given point with using ladder
 	pub fn mul_scalar(&self, le_bytes: [u8; 32]) -> Self {
-		let mut r = Self::zero();
-		let mut exp: EcPoint<W, N, NUM_LIMBS, NUM_BITS, P> = self.clone();
-
-		// Big Endian vs Little Endian
-		let bits = le_bytes.map(|byte| {
-			let mut byte_bits = [false; 8];
-			for i in (0..8).rev() {
-				byte_bits[i] = (byte >> i) & 1u8 != 0
-			}
-			byte_bits
-		});
-		let mut flag = true;
-		// Double and Add operation
-		for bit in bits.flatten() {
-			if *bit {
-				// Addition operation with zero is not working for the add() function because
-				// zero + random value breaks
-				// the algorithm. (Two operands must be distinct and neither of them can be
-				// point at infinity. Otherwise the function returns an erroneous point.) So,
-				// just for the first iteration value assigned manually.
-				if flag {
-					r = exp.clone();
-					flag = false;
-				} else {
-					r = r.add(&exp.clone());
-				}
-			}
-			exp = exp.double();
+		let r_init = Self::to_add();
+		let exp: EcPoint<W, N, NUM_LIMBS, NUM_BITS, P> = self.clone();
+
+		// Converts given input to its bit by Scalar Field's bit size
+		let mut bits = to_bits::<254>(le_bytes);
```

Comment:
Can we use N::NUM_BUTS instead of hardcoding 254

Reply:
Ok, you can leave that one out. We have to find solution to mul_scalar's than passing bytes. I created a task for this: https://github.com/eigen-trust/protocol/issues/158